src/freenet/pluginmanager/PluginManager.java

```diff
@@ -324,12 +308,14 @@ private static String pluginList(ArrayList<PluginInfoWrapper> wrappers) {
 	}
 
 	private String[] getConfigLoadString() {
-		ArrayList<String> v = new ArrayList<String>();
+		List<String> v = new ArrayList<String>();
+		for (PluginInfoWrapper pi : pluginWrappers) {
```

Comment:
Why did you move this before the if(!started) check? I understand why you moved it out of the synchronized block, but if it's not started, why do you want to bother filling in v? Might that have some effect on the timing of wether started is true or not?